src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -728,6 +731,12 @@ namespace OpenMS
 
       for (Size i = 0; i < float_data_arrays_.size(); ++i)
       {
+        if (float_data_arrays_[i].size() != peaks_old)
+        {
+          throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "FloatDataArray[" + String(i) + "] size (" + 
+            String(float_data_arrays_[i].size()) + ") does not match spectrum size (" + String(peaks_old) + ")");
+        }
```

Comment:
there are cases where the float/double arrays may not be of the same lengths as the rest of the spectrum -- maybe we just say then you cannot use `MSSpectrum::select` ?